backends/bmv2/common/action.cpp

```diff
@@ -36,6 +36,29 @@ cstring ActionConverter::jsonAssignment(const IR::Type *type) {
     return "assign"_cs;
 }
 
+namespace {
+/// Invalidates all other headers in a header union except the provided source header.
+/// Has no effect if the parent structure is not a header union.
+void invalidateOtherHeaderUnionHeaders(const IR::Member *sourceHeader,
+                                       const ConversionContext &ctxt, Util::JsonArray *result,
+                                       const IR::StatOrDecl *sourceStatement) {
+    const auto *type = ctxt.typeMap->getType(sourceHeader->expr, true);
+    if (const auto *headerUnionType = type->to<IR::Type_HeaderUnion>()) {
+        for (const auto *field : headerUnionType->fields) {
+            // Ignore the member we are setting to valid.
```

Comment:
maybe remove as the comment is not accurate for the current usage of the function